app/src/main/java/org/piramalswasthya/cho/configuration/PregnantWomanRegistrationDataset.kt

```
@@ -0,0 +1,1314 @@
+            trimester.value = calculateTrimester(weeks)
+        }
+    }
+
+    private fun updateBMI() {
+        val heightValue = height.value?.toDoubleOrNull()
+        val weightValue = weight.value?.toDoubleOrNull()
+
+        if (heightValue != null && heightValue > 0 && weightValue != null && weightValue > 0) {
+            val bmiValue = calculateBMI(weightValue, heightValue)
+            val category = getBMICategory(bmiValue)
+            bmi.value = String.format("%.1f (%s)", bmiValue, category)
```

detekt (1.23.8)

[warning] String.format("%.1f (%s)", bmiValue, category) uses implicitly default locale for string formatting.

(detekt.potential-bugs.ImplicitDefaultLocale)